internal/sinkcores/outputcore/core.go

```diff
@@ -1,22 +1,138 @@
 package outputcore
 
 import (
+	"strings"
 	"time"
 
 	"github.com/sourcegraph/log/internal/encoders"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
-func NewCore(output zapcore.WriteSyncer, level zapcore.LevelEnabler, format encoders.OutputFormat, sampling zap.SamplingConfig) zapcore.Core {
-	core := zapcore.NewCore(
-		encoders.BuildEncoder(format, false),
-		output,
-		level,
-	)
+// Override allows adjusting the log level for specific scopes.
+type Override struct {
+	// Scope matches any zapcore.Entry with the name Scope or the prefix Scope + ".".
+	Scope string
+	// Level is the level to log at for zapcore.Entry's that match Scope.
+	Level zapcore.Level
+}
+
+func NewCore(output zapcore.WriteSyncer, level zapcore.LevelEnabler, format encoders.OutputFormat, sampling zap.SamplingConfig, overrides []Override) zapcore.Core {
+	minOverrideLevel := level
+	for _, o := range overrides {
+		if !minOverrideLevel.Enabled(o.Level) {
+			minOverrideLevel = o.Level
+		}
+	}
+
+	core := &overrideCore{
+		enc: encoders.BuildEncoder(format, false),
+		out: output,
+
+		level:            level,
+		minOverrideLevel: minOverrideLevel,
+
+		overrides: overrides,
+	}
 
 	if sampling.Initial > 0 {
 		return zapcore.NewSamplerWithOptions(core, time.Second, sampling.Initial, sampling.Thereafter)
 	}
 	return core
 }
+
+// override is a zapcore.Core and zapcore.LevelEnabler which additionally
+// will logs entries were allow returns true no matter the level.
+//
+// This does not wrap another core since that would not allow us to override
+// that core's level. It is mostly copy pasta of zapcore.ioCore
```

Comment:
Is there a good way to see what is copy-pasta and what is not?

also nit: let's put this in a separate file in this package

Reply:
Needing to copy-pasta was due to an older version of this code. It is actually cleaner now to just wrap a core like the sampler does. I've updated the code to do that, feels much better.

Also moved to another file.